File: android/src/main/kotlin/io/kumo01/enhanced_preferences/EnhancedPreferencesError.kt
package io.kumo01.enhanced_preferences

class EnhancedPreferencesError(val code: ErrorCode, override val message: String? = null) :
        Throwable()

enum class ErrorCode {
    INVALID_ARGUMENT,
    REFERENCE_ERROR,
    ILLEGAL_ACCESS,
    UNKNOWN_ERROR,
}

File: darwin/enhanced_preferences/Sources/enhanced_preferences/EnhancedPreferencesError.swift
import Foundation

enum EnhancedPreferencesError: Error, LocalizedError {
    case invalidArgument(message: String)
    case referenceError(message: String)
    case illegalAccess(message: String)
    case unknownError(message: String)

    var code: String {
        switch self {
        case .invalidArgument:
            return "INVALID_ARGUMENT"
        case .referenceError:
            return "REFERENCE_ERROR"
        case .illegalAccess:
            return "ILLEGAL_ACCESS"
        case .unknownError:
            return "UNKNOWN_ERROR"
        }
    }

    var errorDescription: String? {
        switch self {
        case .invalidArgument(let message):
            return message
        case .referenceError(let message):
            return message
        case .illegalAccess(let message):
            return message
        case .unknownError(let message):
            return message
        }
    }
}
